examples/ota-provider-app/ota-provider-common/OTAProviderExample.cpp

```diff
@@ -129,47 +131,92 @@ EmberAfStatus OTAProviderExample::HandleQueryImage(chip::app::CommandHandler * c
                                                    const chip::app::ConcreteCommandPath & commandPath,
                                                    const QueryImage::DecodableType & commandData)
 {
-    // TODO: add confiuration for returning BUSY status
-
     OTAQueryStatus queryStatus = OTAQueryStatus::kNotAvailable;
     OTAProviderExample::DeviceSoftwareVersionModel candidate;
-    bool otaAvailable                     = false;
     uint32_t newSoftwareVersion           = 0;
     const char * newSoftwareVersionString = nullptr;
     const char * otaFilePath              = nullptr;
-    bool userConsentNeeded                = false;
     uint8_t updateToken[kUpdateTokenLen]  = { 0 };
     char strBuf[kUpdateTokenStrLen]       = { 0 };
     char uriBuf[kUriMaxLen]               = { 0 };
+    uint32_t delayedActionTimeSec         = mDelayedActionTimeSec;
+    bool requestorCanConsent              = commandData.requestorCanConsent.ValueOr(false);
+    QueryImageResponse::Type response;
 
-    // This use-case is a subset of the ota-candidates-file option.
-    // Can be removed once all other platforms (ESP, etc.)
-    // start using the ota-candidates-file method.
-    if (strlen(mOTAFilePath)) // If OTA file is directly provided
-    {
-        otaAvailable       = true;
-        newSoftwareVersion = commandData.softwareVersion + 1; // This implementation will always indicate that an update is
-                                                              // available (if the user provides a file).
-        newSoftwareVersionString = "Example-Image-V0.1";
-        otaFilePath              = mOTAFilePath;
-    }
-    else if (!mCandidates.empty()) // If list of OTA candidates is supplied instead
+    switch (mQueryImageBehavior)
     {
-        otaAvailable = SelectOTACandidate(commandData.vendorId, commandData.productId, commandData.softwareVersion, candidate);
-        if (otaAvailable)
+    case kRespondWithUnknown:
+        // This use-case is a subset of the ota-candidates-file option.
+        // Can be removed once all other platforms (ESP, etc.)
+        // start using the ota-candidates-file method.
+        if (strlen(mOTAFilePath)) // If OTA file is directly provided
         {
-            newSoftwareVersion       = candidate.softwareVersion;
-            newSoftwareVersionString = candidate.softwareVersionString;
-            otaFilePath              = candidate.otaURL;
+            newSoftwareVersion = commandData.softwareVersion + 1; // This implementation will always indicate that an update is
+                                                                  // available (if the user provides a file).
+            newSoftwareVersionString = "Example-Image-V0.1";
+            otaFilePath              = mOTAFilePath;
+            queryStatus              = OTAQueryStatus::kUpdateAvailable;
```

Comment:
All of this should be read from the OTA file itself. Please add a TODO.

Having invalid new version strings may cause legitimate validity checks related to version numbering in OTA requestors to be failing.